backend/src/graphql/directive/isAuthorized.ts

```diff
@@ -35,6 +35,7 @@ const isAuthorized: AuthChecker<any> = async ({ context }, rights) => {
   const userRepository = await getCustomRepository(UserRepository)
   try {
     const user = await userRepository.findByPubkeyHex(context.pubKey)
+    context.user = user
```

Comment:
I like to discuss if we want `context.user` to be private and access it with `context.getUser()`.

The Idea behind is that we cannot be sure the user object is set, since it depends on what right is queried in the resolver function header.

Consider the following example:

```
  @Authorized([RIGHTS.SOME_PUBLIC_RIGHT])
  @Mutation(() => Boolean)
  async doSomething(@Arg('peter') peter: number, @Ctx() context: any): Promise<boolean> {
    [...]
    const moderatorUser = context.user
    [...]
```

We would not get an error in this case, since `moderatorUser` will just be `undefined` but will not necessarily break the program.

If we keep it like you implemented it its simple and elegant, but programmers must make sure themselves, that the right queried for also writes an user object to the context.

Alternatively we could define a function in the context(or outside for that matter):

```
context.getUser = () => {
   if(!this.user){
      throw new Error('No user found. Not authorized')
   }
  return this.user
}
```